comps/reranks/neural-speed/reranking_neuralspeed_svc.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import os
+import re
+import time
+from typing import List, Optional
+
+import httpx
+import msgspec
+import requests
+from langchain_core.prompts import ChatPromptTemplate
+from langsmith import traceable
+
+from comps import (
+    CustomLogger,
+    LLMParamsDoc,
+    SearchedDoc,
+    ServiceType,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+
+@register_microservice(
+    name="opea_service@reranking_mosec",
+    service_type=ServiceType.RERANK,
+    endpoint="/v1/reranking",
+    host="0.0.0.0",
+    port=8000,
+    input_datatype=SearchedDoc,
+    output_datatype=LLMParamsDoc,
+)
+@traceable(run_type="reranking")
+@register_statistics(names=["opea_service@reranking_mosec"])
+def reranking(input: SearchedDoc) -> LLMParamsDoc:
+    start = time.time()
+    if input.retrieved_docs:
+        docs = [doc.text for doc in input.retrieved_docs]
+        url = mosec_reranking_endpoint + "/inference"
+        data = {"query": input.initial_query, "docs": docs}
+        resp = requests.post(url, data=msgspec.msgpack.encode(data))
+        response = msgspec.msgpack.decode(resp.content)["scores"]
+
+        best_response_index = max(enumerate(response), key=lambda score: score[1])[0]
+        doc = input.retrieved_docs[best_response_index]
+        if doc.text and len(re.findall("[\u4E00-\u9FFF]", doc.text)) / len(doc.text) >= 0.3:
```

Comment:
Now the rerank microservice will not handle the chat template, please align with https://github.com/opea-project/GenAIComps/blob/main/comps/reranks/tei/reranking_tei.py

Reply:
I‘m actually aligned with https://github.com/opea-project/GenAIComps/tree/main/comps/reranks/langchain-mosec. 
So I better align with TEI right ？

Reply:
Only the chat template part needs to align with TEI, now the RAG chat template is applied in LLM microservice, not in reranking microservice.

Reply:
fixed